Timeout on Tycho 4/5 builds

Tycho 4/5 builds regularly get killed due to hanging tests it seems
needs to be investigated

cc @LorenzoBettini @szarnekow 

https://ci.eclipse.org/xtext/job/xtext/job/cd_tycho40/
https://ci.eclipse.org/xtext/job/xtext/job/cd_tycho50/

I managed to see the hang with your suite.
Twice, it hangs in `closeEditorWithChanges`.
If I close the welcome view at that point, the editor is not closed.
If I manually close the editor, the suite goes on with that test failed at:

```
java.lang.AssertionError: There are not deltas
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.eclipse.xtend.ide.tests.builder.AbstractQueuedBuildDataTest.assertThereAreDeltas(AbstractQueuedBuildDataTest.java:99)
	at org.eclipse.xtend.ide.tests.builder.SeveralEditorsQueuedBuildTest.closeEditorWithChanges(SeveralEditorsQueuedBuildTest.java:130)
```

I can't seem to be able to reproduce it if I manually close the welcome screen when the suite starts;
could you please try as well?

--- a/org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/builder/JavaEditorExtension.xtend
+++ b/org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/builder/JavaEditorExtension.xtend
@@ -99,13 +99,25 @@ class JavaEditorExtension {
 					}
 				}
 			], eventMask)
+
 		producer.apply
-		while (!changed.get) {
-			if (Display.getCurrent() !== null) {
-				while (Display.getDefault().readAndDispatch()) {
-					// process queued ui events
+
+		var display = Display.getCurrent()
+		assertNotNull(display)
+
+		val stop = new AtomicBoolean()
+		display.timerExec(10000, [stop.set(true)])
+
+		while (!changed.get && !stop.get) {
+			if (!display.readAndDispatch()) {
+				if (VERBOSE) {
+					println("Display sleep...")
 				}
+				display.sleep();
 			}
+		}
+		if (!changed.get) {
+			throw new AssertionError("No event has been received")
 		}
 		if (VERBOSE) {
 			println('''end waiting for an element changed event: «eventMask»''')

--- a/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/builder/JavaEditorExtension.java
+++ b/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/builder/JavaEditorExtension.java
@@ -134,13 +134,27 @@ public class JavaEditorExtension {
       };
       JavaCore.addElementChangedListener(_function, eventMask);
       producer.apply();
-      while ((!changed.get())) {
-        Display _current = Display.getCurrent();
-        boolean _tripleNotEquals = (_current != null);
-        if (_tripleNotEquals) {
-          while (Display.getDefault().readAndDispatch()) {
+      Display display = Display.getCurrent();
+      Assert.assertNotNull(display);
+      final AtomicBoolean stop = new AtomicBoolean();
+      final Runnable _function_1 = () -> {
+        stop.set(true);
+      };
+      display.timerExec(10000, _function_1);
+      while (((!changed.get()) && (!stop.get()))) {
+        boolean _readAndDispatch = display.readAndDispatch();
+        boolean _not = (!_readAndDispatch);
+        if (_not) {
+          if ((JavaEditorExtension.VERBOSE).booleanValue()) {
+            InputOutput.<String>println("Display sleep...");
           }
+          display.sleep();
         }
+      }
+      boolean _get = changed.get();
+      boolean _not = (!_get);
+      if (_not) {
+        throw new AssertionError("No event has been received");
       }
       String _xifexpression = null;
       if ((JavaEditorExtension.VERBOSE).booleanValue()) {

